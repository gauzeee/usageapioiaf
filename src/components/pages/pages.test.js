import React from "react";
import BookPage from "./bookPage";
import CharacterPage from "./chracterPage";
import HousePage from "./housePage";
import { shallow } from "enzyme";

describe("Testing <CharacterPage/>", () => {
  const item = shallow(<CharacterPage />);
  describe("State and snap", () => {
    it("CharacterPage have rendered correctly", () => {
      expect(item).toMatchSnapshot();
    });

    it('CharacterPage state "charId" is eq to 130', () => {
      expect(item.state().charId).toEqual(130);
    });

    it('CharacterPage state "error" is false', () => {
      expect(item.state().error).toBeFalsy();
    });
  });
  describe("Handlers tests", () => {
    it("testing onItemSelected", () => {
      item.instance().onItemSelected();
      expect(item.state().charId).toEqual(item.id);
    });
  });
});

describe("Testing <HousePage/>", () => {
  const item = shallow(<HousePage />);
  describe("State and snap", () => {
    it("HousePage have rendered correctly", () => {
      expect(item).toMatchSnapshot();
    });

    it('HousePage state "houseId" is eq to 130', () => {
      expect(item.state().houseId).toEqual(130);
    });

    it('HousePage state "error" is false', () => {
      expect(item.state().error).toBeFalsy();
    });
  });
  describe("Handlers tests", () => {
    it("testing onItemSelected", () => {
      item.instance().onItemSelected();
      expect(item.state().houseId).toEqual(item.id);
    });
  });
});

describe("Testing <BookPage/>", () => {
  const item = shallow(<BookPage />);
  describe("State and snap", () => {
    it("BookPage have rendered correctly", () => {
      expect(item).toMatchSnapshot();
    });
  });
});
