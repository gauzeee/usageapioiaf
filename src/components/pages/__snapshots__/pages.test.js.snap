// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BookPage/> State and snap BookPage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(BookPage) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing <CharacterPage/> State and snap CharacterPage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <ItemDetails
        getData={[Function]}
        itemId={130}
      >
        <Field
          field="gender"
          label="Gender"
        />
        <Field
          field="born"
          label="Born"
        />
        <Field
          field="died"
          label="Died"
        />
        <Field
          field="culture"
          label="Culture"
        />
      </ItemDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <ItemDetails
          getData={[Function]}
          itemId={130}
        >
          <Field
            field="gender"
            label="Gender"
          />
          <Field
            field="born"
            label="Born"
          />
          <Field
            field="died"
            label="Died"
          />
          <Field
            field="culture"
            label="Culture"
          />
        </ItemDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing <HousePage/> State and snap HousePage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HousePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <ItemDetails
        getData={[Function]}
        itemId={130}
      >
        <Field
          field="region"
          label="Region"
        />
        <Field
          field="words"
          label="Words"
        />
        <Field
          field="titles"
          label="Titles"
        />
        <Field
          field="currentLord"
          label="Current Lord"
        />
        <Field
          field="coatOfArms"
          label="Coat of Arms"
        />
      </ItemDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <ItemDetails
          getData={[Function]}
          itemId={130}
        >
          <Field
            field="region"
            label="Region"
          />
          <Field
            field="words"
            label="Words"
          />
          <Field
            field="titles"
            label="Titles"
          />
          <Field
            field="currentLord"
            label="Current Lord"
          />
          <Field
            field="coatOfArms"
            label="Coat of Arms"
          />
        </ItemDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
